tests/test_anonymous_access.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import os
+import random
+import tempfile
+
+import boto3
+import botocore
+import botocore.handlers
+import pytest
+
+from framework.bucket_policies.bucket_policy import BucketPolicyBuilder
+from framework.customizations.marks import tier1
+from framework.ssh_connection_manager import SSHConnectionManager
+from noobaa_sa.exceptions import AccountCreationFailed, AccountStatusQueryFailed
+from noobaa_sa.s3_client import S3Client
+
+log = logging.getLogger(__name__)
+
+
+class TestAnonymousAccess:
+    """
+    Test S3 bucket operations with anonymous access
+    """
+
+    @tier1
+    def test_anonymous_account_management(self, account_manager, linux_user_factory):
+        """
+        Test management of the anonymous account via the NooBaa CLI
+        1. Create new Linux users on the NooBaa server
+        2. Create an anonymous account by providing a uid and gid
+        3. Check that its uid and gid match the ones we provided
+        4. Try creating a second anonymous account and check that it fails
+        5. Update the anonymous account with new uid and gid
+        6. Check that the uid and gid were updated
+        7. Update the anonymous account with a new user name
+        8. Check that the user name was updated
+        9. Delete the anonymous account
+        10. Check that the anonymous account was deleted
+        11. Create an anonymous account by providing a user name
+        12. Check that the created account has the provided user name
+        """
+
+        def _verify_anonymous_account(uid=None, gid=None, username=None):
+            """
+            Verify that the anonymous account has the expected uid, gid, or user name
+            """
+            anon_acc_nsfs_config = account_manager.status(account_name="anonymous")[
+                "nsfs_account_config"
+            ]
+            if username:
+                assert (
+                    anon_acc_nsfs_config["distinguished_name"] == username
+                ), f"Expected user: {username}, got {anon_acc_nsfs_config}"
+            elif uid and gid:
+                assert (
+                    anon_acc_nsfs_config["uid"] == uid
+                    and anon_acc_nsfs_config["gid"] == gid
+                ), f"Expected uid: {uid}, gid: {gid}, got {anon_acc_nsfs_config}"
+            else:
+                raise ValueError(
+                    "Either a valid uid and gid pair or a valid user name must be provided"
+                )
+
```

Comment:
This code should be part of `noobaa_sa/account.py` file inside `def create_anonymous(self, uid=None, gid=None, user=None):` method in case of future reusability if we consider creating anonymous account for other operation

Reply:
This validation is specific for the scenario in this TC's: making sure that values are written and updated as expected, and it should be enough to cover it only in this test.

The kind of validation you might want to add to the creation function should be a basic health check. When needed, we could do it like this:
```
account_manager.create_anonymous(uid=uid_a, gid=gid_a)
assert "nsfs_account_config" in account_manager.status(account_name="anonymous")
```

I believe this request is out of the scope of this PR since we would also need to add a similar health check to the creation of the normal accounts for consistency, so please open a new issue and we'll address it in the future.